### ISSUE TITLE
Tasks not opening action form directly from on a contact's profile and on the tasks list

### ISSUE THREAD

**Issue by kitsao** [medic/cht-core#6323]
**Describe the bug**
In app version `3.8.0` clicking on a task when on a contact's profile doesn't open the task form, rather it takes the user to the tasks list. On tasks list, clicking on the same task doesn't open it immediately, the user has to do some other unrelated action such as click on other tasks, which open as expected, close and open the former for it to open.

**To Reproduce**
[CMMB-KE config](https://github.com/medic/config-cmmb-ke/tree/v3.8.0)
```
instance: https://cmmb-ke.dev.medicmobile.org
user: demo
password: V1s10n@2030
```
Steps to reproduce:
1. Login and navigate to `Test test test Household` > `Test test test`
2. Do a `Monthly Screening` on the `Actions` tab. 
```
Is Test test test pregnant? = Yes
How many antenatal care visits has Test test test done so far during this pregnancy? = 0
```
Complete the form and submit.

3. Notice the `Pregnancy confirmation and FP referral follow up` on the person's profile
4. Click on the task. Notice it takes you to the tasks list under tasks tab
5. Click on the same task. Notice the behavior...it doesn't open the action form
6. Click another task to open. Click back the initial task. Notice the action form opening

**Expected behavior**
Clicking on tasks on a contact's profile or tasks list should open the task action form directly.

**Environment**
- Instance: Any `3.8.0` deployment, affecting `cmmb-ke`, `pih-malawi`, `moh-kenya`
- Browser: Chrome
- Client platform: Linux
- App: Webapp, Android
- Version: 3.8.0

**Additional context**
This has been reproduced [here](https://github.com/medic/cht-core/issues/6287#issuecomment-602559940).
Past works related to #1242, #1368

**Comment by dianabarsan** [comment]
I've already started looking into this in order to confirm it. 
Do we add this to `3.8.1` or bump it to `3.9.0` ?

**Comment by kitsao** [comment]
Let's add it to `3.8.1`

**Comment by garethbowen** [comment]
Added to 3.8.1.

**Comment by dianabarsan** [comment]
Ready for AT on `6323-fix-task-loading-race-condition`

**Comment by kitsao** [comment]
Tested locally, works well @dianabarsan

**Comment by ngaruko** [comment]
LGTM  here as well - Feel free to merge and close @dianabarsan .

**Comment by dianabarsan** [comment]
Merged into `master` and backported to `3.8.x`
